### PR TITLE
Level 06: TDD and Vars

### PR DESCRIPTION
**Pull request by endgame**
This looks fun. Not sure how to get a good negative test (can you fabricate a var you've never seen before?), so I suggested we use the type system and make it really hard to cheat instead.

### CODE REVIEW

File: level06/CoffeeMachineTests.hs [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+  , Ensure $ \oldM _ _ refundCoins -> oldM ^. coins === refundCoins
+  ]
+  where
+    gen :: s Symbolic -> Maybe (g (RefundCoins Symbolic))
+    gen _ = Just $ pure RefundCoins
+
+    exec :: RefundCoins Concrete -> m Int
+    exec _ = C.refund mach
+
+cSavePreference
```
Comment (mankyKitty) [inline level06/CoffeeMachineTests.hs]: delete me? / add more instruction?

**Comment by mankyKitty** [comment]
I don't know of a way to construct a `Var` so I don't think you can fake it.. unless the failing test pre-condition is `null prefMap` ? so you can't run the fail test unless you've cleared/never generated preferences?

File: level06/CoffeeMachineTests.hs (hunk 1 above)
Comment (endgame) [reply level06/CoffeeMachineTests.hs]: BALEETED

**Comment by endgame** [comment]
For the fail test you'd still need a key to look up, and the only way to get that key is to have it come from the exec step of a command, which means I think you need a command that generates a key that won't ever be used.

**Pull request merged**